### PR TITLE
Bumping mircommon, miral, and mirserver sonames for mircookie work

### PR DESCRIPTION
**Pull request by mattkae**

### CODE REVIEW

File: src/miral/regenerate-miral-symbols-map.py [hunk 1]
```diff
@@ -555,11 +555,11 @@ def _report(publish, symbol):
   local: *;
 };
 
-MIRAL_4.1 {
+MIRAL_4.2 {
 global:'''
 
 END_NEW_STANZA = '''
-} MIRAL_4.0;
+} MIRAL_4.1;
```
Comment (AlanGriffiths) [inline src/miral/regenerate-miral-symbols-map.py:562]: We need to clear these symbols from the script and let it fill in the current stanza. See 8e3130d7f30204b6d2545694218dab935f68e8a2 for an example

File: src/server/symbols.map [hunk 2]
```diff
@@ -1,4 +1,4 @@
-MIR_SERVER_2.15 {
+MIR_SERVER_2.16 {
```
Comment (AlanGriffiths) [inline src/server/symbols.map:1]: `MIR_SERVER_2.17` - 2.16 has already been released

File: src/common/symbols.map [hunk 3]
```diff
@@ -1,4 +1,4 @@
-MIR_COMMON_2.8 {
+MIR_COMMON_2.15 {
```
Comment (AlanGriffiths) [inline src/common/symbols.map:1]: `MIR_SERVER_2.17` - 2.15 has already been released

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -2,7 +2,7 @@
 set(MIRPLATFORM_ABI 27)
 
 set(MIRAL_VERSION_MAJOR 4)
-set(MIRAL_VERSION_MINOR 1)
+set(MIRAL_VERSION_MINOR 2)
```
Comment (AlanGriffiths) [inline src/CMakeLists.txt:5]: 5.0, not 4.2

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
Thanks for trying, but you're not done yet!

For the libs you've touched, there need to be changes to libmiral6.symbols

And there are more libs to touch: I *think* that mircore is the only ABI that isn't being broken.

For example, `miroil` depends on `mircommon`. So a process needs to be able to load a consistent versions of both libraries. That is only possible with a bump to the miroil .soname.

The same applies to, for example, platforms

File: src/server/symbols.map (hunk 2 above)
Comment (mattkae) [reply src/server/symbols.map:1]: Cool! Why did it say 2.15? Because we didn't introduce any new symbols on the prev release?

**Comment by mattkae** [comment]
> For the libs you've touched, there need to be changes to libmiral6.symbols

Do I do this part by hand?

> For example, `miroil` depends on `mircommon`. So a process needs to be able to load a consistent versions of both libraries. That is only possible with a bump to the miroil .soname.
> 
> The same applies to, for example, platforms

Kk makes sense!

**Comment by mattkae** [comment]
@AlanGriffiths does that include mirwayland?

File: src/server/symbols.map (hunk 2 above)
Comment (AlanGriffiths) [reply src/server/symbols.map:1]: Exactly - the stanza is the release the symbol was introduce to the current .soname

**Comment by AlanGriffiths** [comment]
> @AlanGriffiths does that include mirwayland?

```plain
target_link_libraries(mirwayland
  PUBLIC
    mircore
    PkgConfig::WAYLAND_SERVER
  PRIVATE
    mircommon
)
```

**Comment by AlanGriffiths** [comment]
> > For the libs you've touched, there need to be changes to libmiral6.symbols
> 
> Do I do this part by hand?

The check-and-update-debian-symbols.py (runs as part of a build) adds missing symbols. So just strip all the symbols and `cmake -B ...`.

File: src/miroil/symbols.map
Comment (AlanGriffiths) [inline src/miroil/symbols.map:1]: There's a regenerate-miroil-symbols-map.py (and build target) too

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
We also need to bump the platform ABIs

- MIR_SERVER_INPUT_PLATFORM_ABI
- MIR_SERVER_GRAPHICS_PLATFORM_ABI

**Comment by RAOF** [comment]
I am confused by the failures here. It looks like miroil symbols *used* to be just ignored (`libmiroil3.symbols` when the library was `libmiroil.so.4`), and the miroil symbols diff is removing and then adding back the same symbols.

I'll have another look on Monday if this still fails. Hopefully I'll be fresher then!

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
I think part of the problem is #3184

**Comment by AlanGriffiths** [comment]
@mattkae leave this a bit. I'm debugging the helper stuff

**Comment by AlanGriffiths** [comment]
@mattkae I have to dive into #3185 now, so good luck. (I've fixed some things, but the proof will be in CI. )

Some useful incantations for running the symbol (re)generation scripts:

    cmake --build cmake-build-release --target regenerate-miral-symbols-map
    cmake --build cmake-build-release --target regenerate-miral-debian-symbols

    cmake --build cmake-build-release --target regenerate-miroil-symbols-map
    cmake --build cmake-build-release --target check-miroil-symbols

File: src/miroil/symbols.map [hunk 5]
```diff
@@ -1,4 +1,4 @@
-MIROIL_2.0 {
+MIROIL_5.0 {
```
[the remark this answers is not in the document]
Comment (mattkae) [reply src/miroil/symbols.map:1]: Why should this be 2.17? Is this supposed to be the mir version, not the miroil version?

[the remark this answers is not in the document]
Comment (AlanGriffiths) [reply src/miroil/symbols.map:1]: When I wrote that I was thinking the mir version is more logical. But the check-miroil-symbols script says otherwise.

**Comment by mattkae** [comment]
@AlanGriffiths Ok i think this is good to go again. The symbols that were bumped are:

- miral
- miroil
- mircommon
- mirserver
- mirwayland
- mir-platform-graphics-*
- mir-platform-rendering
- mir-platform-input-*

Am I missing anything?

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
> Am I missing anything?

```
target_link_libraries(mirplatform
  PRIVATE
    mircommon
    ${MIR_PLATFORM_REFERENCES}
  PUBLIC
    PkgConfig::EPOXY
)
```

So, `libmirplatform` also needs an soname bump

**Pull request merged**